ambry-api/src/main/java/com/github/ambry/config/CloudConfig.java

```diff
@@ -448,8 +448,32 @@ public class CloudConfig {
   @Default("false")
   public final boolean vcrHelixUpdateDryRun;
 
+  /**
+   * Not all nodes are identical; some possess more cores while others fewer.
+   * Therefore, adopting a predetermined count of replication threads is not advisable.
+   * Instead, the number of replication threads should be determined based on the number of cores available.
+   * The scaling factor provided below can assume three potential states: negative, zero, or positive.
+   * A negative value signifies the creation of a number of threads equal to the absolute value of the negative integer.
+   * Zero denotes the absence of thread creation.
+   * A positive value will serve as a multiplier for the number of CPUs to ascertain the number of threads to generate.
+   *
+   * For eg, for a node with 4 cores,
+   *  A cpu_scale of -2.5 will create 2 threads since int(|-2.5|) = 2
+   *  A cpu_scale of +2.5 will create 10 threads since int(2.5 * 4) = 10
+   */
+  public static final String BACKUP_NODE_CPU_SCALE = "backup.node.cpu.scale";
+  public static final float DEFAULT_BACKUP_NODE_CPU_SCALE = -1.0f; // 1 thread
+  // num_cpu-1 threads, prevent a fork bomb
+  public static final float MIN_BACKUP_NODE_CPU_SCALE = 1f - Float.valueOf(Runtime.getRuntime().availableProcessors());
+  public static final float MAX_BACKUP_NODE_CPU_SCALE = 4.0f; // 4 * num_cpus, prevent a fork bomb
+  @Config(BACKUP_NODE_CPU_SCALE)
+  public final float backupNodeCpuScale;
```

Comment:
I recommend keeping this simple. If `a predetermined count of replication threads is not advisable`, then let's just rely on the scale factor and only permit the positive (and 0) values.

Reply:
Also, for replication, I think CPU is not the bottleneck since most of the time threads are waiting on response from network than doing busy work. 

We could still have config to scale up and down threads. Probably remove `CPU` references in config names since people may mistake it to be a CPU bottleneck?

Reply:
@justinlin-linkedin Thanks. The idea behind a negative scale is to allow creating 1, 2 or a known number of threads for debugging and experiments. It does not have to be used and will not be used in production. You can still create 1 or 2 threads using the positive scale, but you'd have compute the fraction 1/num_cpus or 2/num_cpus and so on..

@Arun-LinkedIn cpu is not the bottleneck and this pr is not alleviate any cpu bottlenecks. network still remains the limiting factor, but only if ensure we over-subscribe the cpus. This pr removes the guess work of guessing the num threads. This pr future-proofs the code as well. If we get hosts with more cores, then the num threads automatically goes up without any config change. Conversely, it will also go down if we get smaller hosts. I don't think anyone would confuse cpu with bottleneck.

Reply:
Also I am not really interested in the absolute number of threads. I want to know what is the throughput of if I subscribe the cluster as a whole to 2x, 3x or some X times the num avlb cpus.